modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="gitops-argo-rollouts-architecture-overview_{context}"]
+= Argo Rollouts Architecture Overview
+
+Argo Rollouts support is enabled on a cluster by installing the {gitops-title} Operator and configuring a RolloutManager custom resource (CR) instance. 
+
+The {gitops-title} Operator installs Argo Rollouts into a namespace that contains the RolloutManager custom resource. This step includes the installation of the Argo Rollouts controller and the resources required for handling Argo Rollouts, such as Custom Resources, Roles, RoleBindings, and Configuration data.
+
+The Argo Rollouts controller can be installed in either the namespace-scoped or the cluster-scoped mode. In the cluster-scoped mode, which is the default setting, the Rollout controller oversees resources throughout all namespaces within the cluster. In the namespaced-scoped mode, the Rollout controller monitors resources within the namespace where Rollouts is deployed.
+
+The architecture of Argo Rollouts is structured into resources and components in this overview. The components are used in managing the resources. For example, the `AnalysisRun` controller, which is used to perform analysis for an `AnalysisRun` resource, manages the `AnalysisRun` custom resource. Similarly, the Rollouts controller is used for managing the Rollouts custom resource and the Experiment controller handles the Experiment custom resource.
+
+[id="gitops-resources_{context}"]
+== Resources
+
+Argo Rollouts manages several custom resources (CRs), such as Rollout, AnalysisRun, and Experiment along with Kubernetes resources, such as Services, Ingresses, and Routes for network traffic shaping. These resources are essential for customizing the deployment of applications through the Rollout CR. Argo Rollouts is responsible for managing these custom resources to enable the progressive delivery of application resources.
```

Comment:
Fixed.

Reply:
The dichotomy here of custom resources vs kubernetes resources here may be confusing, as:
- All custom resources are also kubernetes resources (kubernetes resources are a superset of custom resources)
- If what you mean by 'kubernetes resources' is 'built-in' kubernetes resources (such as Service, Ingress, etc), then Route would not be on that list, as Route is very much an OpenShift-specific custom resource

Perhaps this would be better:
```
* *Rollouts-contributed resources*: For example, `Rollout`, `AnalysisRun`, or `Experiment`.
* *Kubernetes networking resources*: For example, `Service`, `Ingress`, or `Route` for network traffic shaping. 
```

'Rollouts-contributed' being resources that are coming from/installed by, the Rollouts operator. Another better term might exist... 'Rollouts-operator resources'? Open to suggestions!